source/core/slang-md5.h

```diff
@@ -70,6 +70,8 @@ namespace Slang
         void update(MD5Context* ctx, String str);
```

Comment:
I would agree with @jsmall-nvidia that we should rename the templated `update` to something else, like `updateSimpleVal` to prevent it from being picked up by a call to `update` with a complex type, e.g. `ComPtr<ISlangBlob>` unintentionally.

Reply:
Have enable_if guarding the general one on line 57 (for enums/arithmetic values), the second template is for Lists which shouldn't get picked up by anything that isn't a List.